### ISSUE TITLE
Impossible to run tests in parallel (maybe if a server listening on port 5000 is running)

### ISSUE THREAD

**Issue by rholy** [rpm-software-management/librepo#64]
If I clone the repository twice:

```sh
git clone https://github.com/Tojaj/librepo.git librepo1
cd librepo1
mkdir build
cd build
cmake ..
make
cd ../..
git clone https://github.com/Tojaj/librepo.git librepo2
cd librepo2
mkdir build
cd build
cmake ..
make
```

and if I run the tests in parallel:
```sh
[librepo1/build]$ make ARGS=-V test
```
```sh
[librepo2/build]$ make ARGS=-V test
```

I get this error:
```
[...]
2: test_read_mirrorlist (tests.test_yum_repo_locating.TestCaseYumRepoLocating) ... ok
2: 
2: ======================================================================
2: FAIL: test_download_with_offline_enabled_04 (tests.test_yum_repo_downloading.TestCaseYumRepoDownloading)
2: ----------------------------------------------------------------------
2: Traceback (most recent call last):
2:   File "/home/rholy/h/librepo1/tests/python/tests/test_yum_repo_downloading.py", line 1620, in test_download_with_offline_enabled_04
2:     self.assertEqual(h.mirrors, EXP_MRS)
2: AssertionError: Lists differ: [u'http://127.0.0.1:5001/yum/s... != [u'http://127.0.0.1:5000/yum/s...
2: 
2: First differing element 0:
2: http://127.0.0.1:5001/yum/static/01/
2: http://127.0.0.1:5000/yum/static/01/
2: 
2: - [u'http://127.0.0.1:5001/yum/static/01/',
2: ?                        ^
2: 
2: + [u'http://127.0.0.1:5000/yum/static/01/',
2: ?                        ^
2: 
2: -  u'http://127.0.0.1:5001/yum/static/01/']
2: ?                        ^
2: 
2: +  u'http://127.0.0.1:5000/yum/static/01/']
2: ?                        ^
2: 
2: 
2: ----------------------------------------------------------------------
2: Ran 105 tests in 5.032s
[...]
```

This becomes a problem if you try e.g. to build the RPMs for two different architectures at the same time:
```sh
export TESTDIR=/tmp/test
curl -o librepo.spec http://pkgs.fedoraproject.org/cgit/librepo.git/plain/librepo.spec?id=f22
mock --root=fedora-22-x86_64 --install /usr/bin/git check-devel cmake expat-devel gcc glib2-devel gpgme-devel libattr-devel libcurl-devel openssl-devel python2-devel python3-devel pygpgme python3-pygpgme python-flask python3-flask python-nose python3-nose pyxattr python3-pyxattr doxygen python-sphinx python3-sphinx
mock --root=fedora-22-x86_64 --chroot -- rm --recursive --force $TESTDIR
mock --root=fedora-22-x86_64 --copyin . $TESTDIR
mock --root=fedora-22-x86_64 --chroot 'ln --symbolic --force /builddir/build "$HOME/rpmbuild"'
mock --root=fedora-22-i386 --install /usr/bin/git check-devel cmake expat-devel gcc glib2-devel gpgme-devel libattr-devel libcurl-devel openssl-devel python2-devel python3-devel pygpgme python3-pygpgme python-flask python3-flask python-nose python3-nose pyxattr python3-pyxattr doxygen python-sphinx python3-sphinx
mock --root=fedora-22-i386 --chroot -- rm --recursive --force $TESTDIR
mock --root=fedora-22-i386 --copyin . $TESTDIR
mock --root=fedora-22-i386 --chroot 'ln --symbolic --force /builddir/build "$HOME/rpmbuild"'
```
```sh
mock --root=fedora-22-x86_64 --chroot "cd $TESTDIR && utils/make_rpm.sh ."
```
```sh
mock --root=fedora-22-i386 --chroot "cd $TESTDIR && utils/make_rpm.sh ."
```
(BTW, the reproducer can be much simpler with https://github.com/Tojaj/librepo/pull/61 ;-) )

I didn't test it but maybe it isn't even needed to run the tests twice. Maybe it's enough to start a server listening on the port 5000.

But feel free to close this if you think it's not worth fixing.

**Comment by cgwalters** [comment]
For ostree I wrote: https://git.gnome.org/browse/ostree/tree/src/ostree/ot-builtin-trivial-httpd.c

Which dynamically allocates an unused port and prints it.

**Comment by rholy** [comment]
If I understand the error correctly, I think that Flask (the server used in tests) automatically changes the port but the tests does not expect this change.

**Comment by Tojaj** [comment]
Hi Radek,
support for dynamical port choosing was added some time ago (22th Dec 2014) by this commit:

https://github.com/rpm-software-management/librepo/commit/47a50f3922ef215c4bbf36b4597e130a7cc30f1f

This looks like a bug in a single test (a test for LRO_OFFLINE - a feature I added recently).
I'm going to investigate this a fix it asap.

Thanks for report

**Comment by Tojaj** [comment]
Fixed: cd6b8b72c8ce085d42c908a388438404c11cdf13
